### PR TITLE
Add updated doxygen configuration

### PR DESCRIPTION
**Pull request by gordonwang0**
*Description of changes:*
Adds a doxygen configuration file to generate the main page for documentation.

To view, run from SDK's root directory:
`
$ doxygen doc/config/main
`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -70,13 +70,13 @@ Steps:
     * `platform` - Platform specific files for timer, TLS and threading layers
     * `samples` - The sample applications
     * `src` - The AWS IoT SDK source files
-    * `tests` - Contains tests for verifying that the SDK is functioning as expected. More information can be found [here](https://github.com/aws/aws-iot-device-sdk-embedded-c/blob/master/tests/README.md)
- * View further information on how to use the SDK in the Readme file for samples that can be found [here](https://github.com/aws/aws-iot-device-sdk-embedded-c/blob/master/samples/README.md)
+    * `tests` - Contains tests for verifying that the SDK is functioning as expected. More information can be found [here](https://github.com/aws/aws-iot-device-sdk-embedded-c/blob/master/tests/README.md/)
```
Comment (dcgaws) [inline README.md]: This added trailing slashes to the URL. Intended?

File: README.md [hunk 2]
```diff
@@ -70,13 +70,13 @@ Steps:
...
+ * View further information on how to use the SDK in the Readme file for samples that can be found [here](https://github.com/aws/aws-iot-device-sdk-embedded-c/blob/master/samples/README.md/)
 
 Also, for a guided example on getting started with the Thing Shadow, visit the AWS IoT Console's [Interactive Guide](https://console.aws.amazon.com/iot).
 
 ## Porting to different platforms
-As Embedded devices run on different Real Time Operating Systems and TCP/IP stacks, it is one of the important design goals for the Device SDK to keep it portable. Please refer to the [porting guide](https://github.com/aws/aws-iot-device-sdk-embedded-C/blob/master/PortingGuide.md) to get more information on how to make this SDK run on your devices (i.e. micro-controllers).
+As Embedded devices run on different Real Time Operating Systems and TCP/IP stacks, it is one of the important design goals for the Device SDK to keep it portable. Please refer to the [porting guide](https://github.com/aws/aws-iot-device-sdk-embedded-C/blob/master/PortingGuide.md/) to get more information on how to make this SDK run on your devices (i.e. micro-controllers).
```
Comment (dcgaws) [inline README.md]: "real-time operating systems" "microcontrollers"

File: README.md [hunk 3]
```diff
@@ -98,7 +98,7 @@ The 2.x branch makes several changes to the SDK. This section provides informati
     | Publish | ```IoT_Error_t aws_iot_mqtt_publish(MQTTPublishParams *pParams);``` | ```IoT_Error_t aws_iot_mqtt_publish(AWS_IoT_Client *pClient, const char *pTopicName, uint16_t topicNameLen, IoT_Publish_Message_Params *pParams);``` |
     | Disconnect | ```IoT_Error_t aws_iot_mqtt_disconnect(void);``` | ```IoT_Error_t aws_iot_mqtt_disconnect(AWS_IoT_Client *pClient);``` |
 
-You can find more information on how to use the new APIs in the Readme file for samples that can be found [here](https://github.com/aws/aws-iot-device-sdk-embedded-c/blob/master/samples/README.md)
+You can find more information on how to use the new APIs in the Readme file for samples that can be found [here](https://github.com/aws/aws-iot-device-sdk-embedded-c/blob/master/samples/README.md/)
```
Comment (dcgaws) [inline README.md]: "README file"

File: README.md (hunk 1 above)
Comment (gordonwang0) [reply README.md]: Yes.  Strangely, when doxygen sees a link ending in .md, it tries to find a .md file on the filesystem to link (which won't be found).  But if it ends in a /, doxygen recognizes it as a URL and links correctly to that.

File: README.md (hunk 2 above)
Comment (gordonwang0) [reply README.md]: fixed

File: README.md (hunk 3 above)
Comment (gordonwang0) [reply README.md]: fixed

**Pull request merged**